File: 动态规划/动归-oj/股票/121.最佳时机/最佳时机.java
//leetcode121
//https://leetcode.cn/problems/best-time-to-buy-and-sell-stock/
class Solution {
    public int maxProfit(int[] prices) {
     int num=prices.length;
     int minprice=Integer.MAX_VALUE;//整形最大值
     int maxprofit=0;
     for(int i=0;i<num;i++)
     {
         if(prices[i]<minprice)
         {
             minprice=prices[i];
         }
         else if(prices[i]-minprice>maxprofit)
         {
             maxprofit=prices[i]-minprice;
         }

     }
     return maxprofit;
    }
}

File: 动态规划/动归-oj/股票/121.最佳时机/最佳时机.cpp
//leetcode121
//https://leetcode.cn/problems/best-time-to-buy-and-sell-stock/
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int inf=1e9;//较大数
        int minprice=inf,maxprofit=0;
        for(int price:prices)//遍历数组
        {
            maxprofit=max(maxprofit,price-minprice);//minprice为当前天数之前最小价格
            minprice=min(minprice,price);
        }
        return maxprofit;
    }
};